openpgp/packet/public_key.go

```diff
@@ -313,6 +314,23 @@ func (pk *PublicKey) setFingerprintAndKeyId() {
 	}
 }
 
+func (pk *PublicKey) checkV6Compatibility() error {
+	// Implementations MUST NOT accept or generate version 6 key material using the deprecated OIDs.
+	switch pk.PubKeyAlgo {
+	case PubKeyAlgoECDH:
+		curveInfo := ecc.FindByOid(pk.oid)
+		if curveInfo == nil {
+			return errors.UnsupportedError(fmt.Sprintf("unknown oid: %x", pk.oid))
+		}
+		if curveInfo.GenName == ecc.Curve25519GenName {
+			return errors.StructuralError("cannot generate v6 key with deprecated OID: Curve25519Legacy")
+		}
+	case PubKeyAlgoEdDSA:
+		return errors.StructuralError("cannot generate v6 key with deprecated OID: Ed25519Legacy")
```

Comment:
Bit nitpicky but for this one it's the entire algorithm that's deprecated, not just the OID:
```suggestion
		return errors.StructuralError("cannot generate v6 key with deprecated algorithm: EdDSALegacy")
```